src/Symfony/Component/Messenger/Bridge/MongoDB/Transport/Connection.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\MongoDB\Transport;
+
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Collection;
+use MongoDB\Driver\Cursor;
+use MongoDB\Driver\Exception\RuntimeException as DriverException;
+use MongoDB\Driver\Session;
+use MongoDB\Driver\WriteConcern;
+use MongoDB\Model\BSONDocument;
+use MongoDB\Operation\FindOneAndUpdate;
+use Symfony\Component\Clock\Clock;
+use Symfony\Component\Clock\ClockInterface;
+use Symfony\Component\Messenger\Bridge\MongoDB\Extension\DocumentEnhancer;
+use Symfony\Component\Messenger\Bridge\MongoDB\Stamp\MongoDBSessionStamp;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\TransportException;
+
+/**
+ * @internal
+ *
+ * @author Alessandro Lai <alessandro.lai85@gmail.com>
+ */
+final class Connection
+{
+    /** @var DocumentEnhancer[] */
+    private array $documentEnhancers = [];
+
+    private readonly string $uniqueId;
+
+    public function __construct(
+        private readonly Collection $collection,
+        private readonly string $queueName,
+        private readonly int $redeliverTimeout,
+        private ClockInterface $clock = new Clock(),
+    ) {
+        $this->uniqueId = uniqid('consumer_', true);
+    }
+
+    public function addDocumentEnhancer(DocumentEnhancer $enhancer): void
```

Comment:
Can we make the class immutable instead, by injecting an iterable of document enhancers in the constructor instead ?